File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/String.SaveAs.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.IO

Public Module Extensions_529

	''' <summary>
	'''     A string extension method that save the string into a file.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="fileName">Filename of the file.</param>
	''' <param name="append">(Optional) if the text should be appended to file file if it's exists.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub SaveAs(this As String, fileName As String, Optional append As Boolean = False)
		Using tw As TextWriter = New StreamWriter(fileName, append)
			tw.Write(this)
		End Using
	End Sub

	''' <summary>
	'''     A string extension method that save the string into a file.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="file">The FileInfo.</param>
	''' <param name="append">(Optional) if the text should be appended to file file if it's exists.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub SaveAs(this As String, file As FileInfo, Optional append As Boolean = False)
		Using tw As TextWriter = New StreamWriter(file.FullName, append)
			tw.Write(this)
		End Using
	End Sub
End Module

File: src/Z.Core/System.String/String.SaveAs.cs
// Description: C# Extension Methods Library to enhances the .NET Framework by adding hundreds of new methods. It drastically increases developers productivity and code readability. Support C# and VB.NET
// Website & Documentation: https://github.com/zzzprojects/Z.ExtensionMethods
// Forum: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License: https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: http://www.zzzprojects.com/
// Copyright © ZZZ Projects Inc. 2014 - 2016. All rights reserved.
using System.IO;

public static partial class Extensions
{
    /// <summary>
    ///     A string extension method that save the string into a file.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="fileName">Filename of the file.</param>
    /// <param name="append">(Optional) if the text should be appended to file file if it's exists.</param>
    public static void SaveAs(this string @this, string fileName, bool append = false)
    {
        using (TextWriter tw = new StreamWriter(fileName, append))
        {
            tw.Write(@this);
        }
    }

    /// <summary>
    ///     A string extension method that save the string into a file.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="file">The FileInfo.</param>
    /// <param name="append">(Optional) if the text should be appended to file file if it's exists.</param>
    public static void SaveAs(this string @this, FileInfo file, bool append = false)
    {
        using (TextWriter tw = new StreamWriter(file.FullName, append))
        {
            tw.Write(@this);
        }
    }
}